src/SFML/Audio/SoundFileWriterOpus.cpp

```diff
@@ -0,0 +1,259 @@
+{
+    std::string extension = filename.substr(filename.find_last_of(".") + 1);
+    std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+
+    return extension == "opus";
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::SoundFileWriterOpus() :
+m_channelCount (0),
+m_file         (),
+m_ogg          (),
+m_opus         (NULL),
+m_packageNumber(0)
+{
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::~SoundFileWriterOpus()
+{
+    close();
+}
+
+
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::open(const std::string& filename, unsigned int sampleRate, unsigned int channelCount)
+{
+    // Save the channel count
+    m_channelCount = channelCount;
+    m_sampleRate = sampleRate;
+    std::srand(std::time(0));
+
+    // Initialize the ogg/opus stream
+    if (ogg_stream_init(&m_ogg, std::rand()) == -1)
+    {
+        err() << "Stream init of ogg/opus file \"" << filename << "\" failed" << std::endl;
+        close();
+        return false;
+    }
+    int status = 0;
+    m_opus = opus_encoder_create(sampleRate, channelCount, OPUS_APPLICATION_AUDIO, &status);
+    if (status != OPUS_OK)
+    {
+        err() << "Failed to write ogg/opus file \"" << filename << "\"" << std::endl;
+        if (status == OPUS_BAD_ARG)
+            err() << "Possible wrong sample rate, allowed are 8000, 12000, 16000, 24000, or 48000 Hz." << std::endl;
+        close();
+        return false;
+    }
+
+    m_file.open(filename.c_str(), std::ios::binary);
+    if (!m_file)
+    {
+        err() << "Failed to write ogg/opus file \"" << filename << "\" (cannot open file)" << std::endl;
+        close();
+        return false;
+    }
+
+    // Set bitrate (VBR is default)
+    opus_encoder_ctl(m_opus, OPUS_SET_BITRATE(128000));
+
+    // Create opus header
+    unsigned char headerData[19];
+    memset(static_cast<void*>(headerData), 0, 19);
+    memcpy(static_cast<void*>(headerData), "OpusHead", 8);
+    headerData[8] = 1; // Version
+    headerData[9] = channelCount;
+    writeUint32(headerData + 12, static_cast<Uint32>(sampleRate));
+    headerData[18] = channelCount > 8 ? 255 : (channelCount > 2); // Mapping family
+
+    // Map opus header to ogg packet
+    ogg_packet op;
+    op.packet = headerData;
+    op.bytes = 19;
+    op.b_o_s = 1;
+    op.e_o_s = 0;
+    op.granulepos = 0;
+    op.packetno = m_packageNumber++;
+
+    // Write the header packet to the ogg stream
+    ogg_stream_packetin(&m_ogg, &op);
+    flushBlocks();
+
+    // Create comment header, needs to be in a new page
+    const char* opusVersion = opus_get_version_string();
+    Uint32 opusVersionLength = strlen(opusVersion);
+    const int commentLength  = 8 + 4 + opusVersionLength + 4;
+
+    unsigned char commentData[commentLength];
+
+    // Magic bytes
+    memcpy(static_cast<void*>(commentData), "OpusTags", 8);
```

Comment:
> std::memcpy -- and conversion to void* can be implicit.
Also in other places...

Ok I will add the namespace, my intention for the cast was to avoid compiler warnings (-Wpedantic).